src/aws/osml/model_runner/inference/sm_detector.py

```diff
@@ -116,16 +115,14 @@ def find_features(self, payload: BufferedReader, metrics: MetricsLogger) -> Feat
                 f"Unable to get detections from model - HTTP Status Code: {http_status_code}, Error Code: {error_code}"
             )
             logger.exception(ce)
+            raise ce
         # If there was an error parsing the models output
         except JSONDecodeError as de:
-            self.error_count += 1
             if isinstance(metrics, MetricsLogger):
                 metrics.put_metric(MetricLabels.ERRORS, 1, str(Unit.COUNT.value))
             logger.error("Unable to decode response from model.")
             logger.exception(de)
-
-        # Return an empty feature collection if the process errored out
-        return FeatureCollection([])
```

Comment:
If we are not returning an empty collection, I would suggest updating the method return types and docstring as well.  

I feel like we should return an emptycollection.

Reply:
It does return a feature collection if the model returns one.

Reply:
I think this is ok since all of the caught exceptions are re-raised here. The docstring lists the correct output type which should be returned for all non-exception paths and it also has 2 new raises clauses to document that exceptions can be thrown by this method. 

The original return of an empty feature collection is because in the prior iterations of this Detector class some errors were handled inside this class. That changed now and the explicit assumption is that the caller must decide what to do anytime a call to find_features throws an exception. The process_tiles() method on the TileWorker already has a try/catch block for all Exceptions that meets this contract so it seems reasonable to just let it handle the error logic.

If we wanted to tighten this up we could do more error checking on what the model returns. Strictly speaking not every result from a geojson.loads call will be a FeatureCollection so we're not really guaranteeing that the result type is correct in the nominal path. A model could return some other valid geojson construct that would not trigger a parse error and we would try to return it from our detector.